test/integration/publishing_event_processor_test.rb

```diff
@@ -0,0 +1,64 @@
+require 'integration_test_helper'
+require 'bunny-mock'
+require 'govuk_index/publishing_event_processor'
+require 'govuk_message_queue_consumer'
+
+class GovukIndex::PublishingEventProcessorTest < IntegrationTest
+  def setup
+    bunny_mock = BunnyMock.new
+    channel = bunny_mock.start.channel
+
+    consumer = GovukMessageQueueConsumer::Consumer.new(
+      queue_name: "bigwig.test",
+      processor: GovukIndex::PublishingEventProcessor.new("govuk_test"),
+      rabbitmq_connection: bunny_mock
+    )
+
+    @queue = channel.queue("bigwig.test")
+    consumer.run
+  end
+
+  def test_should_save_new_document_to_elasticsearch
+    fixture_file = File.expand_path("../fixtures/publishing_api_payload.json", __FILE__)
+    payload = JSON.parse(File.read(fixture_file))
```

Comment:
The consumer is setup to do JSON parsing internally, as discuss previously I believe that we should be passing in JSON data so that we match the production implementation as closely as possible. I think this can be done using:

```
@queue.publish(json_payload, headers: { content_type: "application/json" })
```